### DISCUSSION TITLE
Kubernetes Network Emulation (KNE) provider

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#582] [Ideas]
[KNE](https://github.com/openconfig/kne) is a recent development in open source networking, it can build a topology of network devices on top of a Kubernetes cluster

Example: https://github.com/openconfig/kne/blob/main/examples/srlinux/2node-srl-ixr6-with-oc-services.pbtxt

Early days, but it should be fairly straightforward to generate a compatible config file from Netlab, and deploy/configure labs that way

**Comment by ipspace** [comment]
If you feel like creating a whole new provider, please go for it. Reading the docs, I got the impression it's all a bit Rube Goldberg concoction at the moment (many moving parts, tons of things to install...)
